main.go

```diff
@@ -208,6 +210,33 @@ func populateArgs(c *cli.Context) (
 	return
 }
 
+func callListRecursive(mountPoint string) (err error) {
+	logger.Debugf("Started recursive metadata-prefetch of directory: \"%s\" ...", mountPoint)
+	numItems := 0
+	err = filepath.WalkDir(mountPoint, func(path string, d fs.DirEntry, err error) error {
+		if err == nil {
+			numItems++
+			return err
+		}
+		if d == nil {
+			return fmt.Errorf("got error walking: path=\"%s\" does not exist, error = %w", path, err)
+		}
+		return fmt.Errorf("got error walking: path=\"%s\", dentry=\"%s\", isDir=%v, error = %w", path, d.Name(), d.IsDir(), err)
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed in recursive metadata-prefetch of directory: \"%s\"; error = %w", mountPoint, err)
+	}
+
+	logger.Debugf("... Completed recursive metadata-prefetch of directory: \"%s\". Number of items discovered: %v", mountPoint, numItems)
+
+	return nil
+}
+
+func isDynamicMount(bucketName string) bool {
```

Comment:
We do similar check in mount.go too. Can we export it and use it at both the places.

Reply:
done